homeassistant/components/fastdotcom/__init__.py

```diff
@@ -42,51 +43,29 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
                 data=config[DOMAIN],
             )
         )
+    await async_setup_services(hass)
```

Comment:
Why is this not in `async_setup_entry()`?

Reply:
It was [pointed out by Martin](https://github.com/home-assistant/core/pull/104839#discussion_r1429269526) it should be in `async_setup`. What is the exact difference in this case?

Reply:
Don't understand why it should be here but maybe @MartinHjelmare can clarify why it shouldn't be in `_async_finish_startup()`
In current case we could have the service but a non working integration which seems to make no sense to me.

Reply:
We've decided that integration services should not be removed depending on the config entry state. It's a better user experience to be able to call the service and get an explanation why the service doesn't work than just get the missing service error.

Reply:
Agree about not removing them but should they really be created without any related config entry eg create based on first `setup_entry`?

As it's now done everyone gets this service regardless if they will ever use this integration

Reply:
`async_setup` isn't called without asking to set up the integration (and a config entry) at least once. If the user disables all config entries for an integration and restarts Home Assistant the `async_setup` function for the integration won't be called so the service won't be registered.

Reply:
Thanks. Makes sense but wasn't sure what triggering it to run or not 👍 

Still I would argue to put it in `setup_entry` to avoid keeping `CONFIG_SCHEMA` around.

Reply:
Putting it in `async_setup_entry` makes it fragile as the config entry setup can fail before the service is registered unless care is taken to do it first. It's more clear to put it in `async_setup` as we want to register the service if the integration should be setup regardless of the config entry state.

Reply:
I see what you mean so that makes sense to make it safer 👍